crates/core/keys/src/address.rs

```diff
@@ -319,7 +342,29 @@ impl std::str::FromStr for Address {
     type Err = anyhow::Error;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        if s.starts_with(bech32str::compat_address::BECH32_PREFIX) {
+        if s.starts_with(TRANSPARENT_ADDRESS_BECH32_PREFIX) {
+            let dzero = Diversifier([0u8; 16]);
+
+            let pk_dzero_bytes: [u8; 32] =
+                bech32str::decode(s, TRANSPARENT_ADDRESS_BECH32_PREFIX, bech32str::Bech32)?
+                    .try_into()
+                    .map_err(|bytes: Vec<u8>| {
+                        anyhow::anyhow!("wrong length {}, expected 32", bytes.len())
+                    })?;
+            let pk_dzero = ka::Public(pk_dzero_bytes);
+
+            let ck_id = fmd::ClueKey([0u8; 32]);
+
+            let address = Self::from_components(dzero, pk_dzero, ck_id)
+                .ok_or_else(|| anyhow::anyhow!("could not reconstruct transparent address"))?;
+
+            // Verify this is a valid transparent address, bailing if not
+            if address.encode_as_transparent_address().is_none() {
+                return Err(anyhow::anyhow!("invalid transparent address components"));
+            }
```

Comment:
don't we know this will always be the case because we just constructed `d`, `ck_d`?

Reply:
yep.. yep we do